core-bundle/src/Twig/Inheritance/DynamicExtendsTokenParser.php

```diff
@@ -73,12 +74,14 @@ private function handleContaoExtends(Node $expr, TokenStream $stream): void
         TokenParserHelper::traverseConstantExpressions(
             $expr,
             function (Node $node) use ($stream): void {
-                if (null === ($shortName = TokenParserHelper::getContaoTemplate($node->getAttribute('value')))) {
+                $parts = ContaoTwigUtil::parseContaoName($node->getAttribute('value'));
+
+                if ('Contao' !== ($parts[0] ?? null)) {
```

Comment:
Why this check? If `$parts` is not `null`, `$parts[0]` will always be `Contao`, won't it?

Reply:
It could be `Contao` but also `Contao_Foo`, `Contao_App` and so on.

Reply:
Should we need more name parsing in the future, we can refactor this to return an object with getters instead. I'd keep it as internal first anyway.